book/sources/exex/tracking-state/src/bin/1.rs

```diff
@@ -0,0 +1,58 @@
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{ready, Context, Poll},
+};
+
+use futures_util::StreamExt;
+use reth::api::FullNodeComponents;
+use reth_exex::{ExExContext, ExExEvent, ExExNotification};
+use reth_node_ethereum::EthereumNode;
+use reth_tracing::tracing::info;
+
+struct MyExEx<Node: FullNodeComponents> {
+    ctx: ExExContext<Node>,
+}
+
+impl<Node: FullNodeComponents> Future for MyExEx<Node> {
+    type Output = eyre::Result<()>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let this = self.get_mut();
+
+        while let Some(result) = ready!(this.ctx.notifications.poll_next_unpin(cx)) {
+            let notification = result?;
```

Comment:
and in other places too

```suggestion
        while let Some(result) = ready!(this.ctx.notifications.try_next().poll_unpin(cx))? {
```

Reply:
Done